openedx/core/djangoapps/course_roles/api.py

```diff
@@ -0,0 +1,55 @@
+"""
+Helpers for the course roles app.
+"""
+from __future__ import annotations
+
+from django.contrib.auth.models import AnonymousUser, User  # lint-amnesty, pylint: disable=imported-auth-user
+from django.db.models import Q
+from opaque_keys.edx.keys import CourseKey
+
+from edx_django_utils.cache import RequestCache
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.djangoapps.course_roles.models import UserRole
+from openedx.core.djangoapps.course_roles.data import CourseRolesPermission
+
+
+def get_all_user_permissions_for_a_course(
+    user: User | AnonymousUser, course_key: CourseKey
+) -> set[CourseRolesPermission]:
+    """
+    Get all of a user's permissions for a course,
+    including, if applicable, organization-wide permissions
+    and instance-wide permissions.
+    """
+    if isinstance(user, AnonymousUser):
+        return set()
+    if not isinstance(course_key, CourseKey):
+        raise TypeError('course_key must be a CourseKey')
+    if not isinstance(user, User):
+        raise TypeError('user must be a User')
+    cache = RequestCache("course_roles")
+    cache_key = f"all_user_permissions_for_course:{user.id}:{course_key}"
+    cached_response = cache.get_cached_response(cache_key)
+    if cached_response.is_found:
+        return cached_response.value
+    if not CourseOverview.course_exists(course_key):
+        raise ValueError('course does not exist')
```

Comment:
Please add the offending CourseKey to this ValueError so that it says something like `course does not exist: library-v1:Axim+100` in the error message.

This might also be worth a error log message.